### PR TITLE
test: Update GHA thumbprint for OIDC

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #N/A 

**Description**

GHA [updated their thumbprint](https://github.com/aws-actions/configure-aws-credentials/issues/357#issuecomment-1605290219) for OIDC provider token communication using the `configure-aws-credentials` action

**How was this change tested?**

`/karpenter snapshot`

**Does this change impact docs?**
- [] Yes, PR includes docs updates
- [] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/gha/cloudformation/iam_cloudformation.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ Resources:
       ClientIdList:
         - sts.amazonaws.com
       ThumbprintList:
-        - 6938fd4d98bab03faadb97b34396831e3780aea1
+        - f879abce0008e4eb126e0097e46620f5aaae26ad
```
Comment (njtran) [inline test/gha/cloudformation/iam_cloudformation.yaml:21]: Shouldn't this be both of them? https://github.blog/changelog/2023-06-27-github-actions-update-on-oidc-integration-with-aws/ and shouldn't the second one be `1c58a3a8518e8759bf075b76b750d4f2df264fcd`?

Comment (jonathan-innis) [reply test/gha/cloudformation/iam_cloudformation.yaml]: Ah, good find. I didn't see that other blogpost

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

File: test/gha/cloudformation/iam_cloudformation.yaml (hunk 1 above)
Comment (njtran) [reply test/gha/cloudformation/iam_cloudformation.yaml:26]: This link only has the first two, where are you getting the last one from?

Comment (jonathan-innis) [reply test/gha/cloudformation/iam_cloudformation.yaml]: Just removed it. Seems like there is another cert that was in that discussion thread but I'm not sure that it was neccesary.

**Review by njtran — APPROVED** [review]

**Review by jonathan-innis — COMMENTED** [review]
/karpenter snapshot

**Pull request merged**